packages/editor-ui/src/modules/settings.ts

```diff
@@ -24,6 +25,10 @@ const module: Module<ISettingsState, IRootState> = {
 			smtpSetup: false,
 		},
 		templatesEndpointHealthy: false,
+		api: {
+			enabled: false,
+			key: undefined,
```

Comment:
you really wanna keep the key in the store? What about fetching it in the`SettingsApiView` on mount and it is gone on unmount?

Reply:
Updated to no longer use the store for it.

Reply:
Would this work?: If we fetch and receive `{ apiKey: null }` from the BE then we know it is deletable and can trigger the telemetry on the action